### ISSUE TITLE
BUG: f2py fails on "hiperseis" program

### ISSUE THREAD

**Issue by baerbock** [numpy/numpy#14232]
I'm reporting here since f2py is part of numpy, but this is ONLY f2py-related!

As reported in https://github.com/GeoscienceAustralia/hiperseis/issues/121 f2py fails on
https://github.com/GeoscienceAustralia/ellip-corr/blob/master/ellip/ellipcorr.f

```
Reading fortran codes...
	Reading file 'ellip-corr/ellip/ellipcorr.f' (format:fix,strict)
Post-processing...
	Block: ellipcorr
			Block: ellipticity_corr
{}
In: :ellipcorr:ellip-corr/ellip/ellipcorr.f:phase_alias
vars2fortran: No typespec for argument "delta".
{}
In: :ellipcorr:ellip-corr/ellip/ellipcorr.f:phase_alias
vars2fortran: No typespec for argument "ip".
			Block: phase_alias
			Block: lnblk
Post-processing (stage 2)...
Building modules...
	Building module "ellipcorr"...
		Constructing wrapper function "ellipticity_corr"...
		  tcor = ellipticity_corr(phase,edist,edepth,ecolat,azim)
		Constructing wrapper function "phase_alias"...
		  phase_alias(phase,delta,ip)
		Creating wrapper for Fortran function "lnblk"("lnblk")...
		Constructing wrapper function "lnblk"...
		  lnblk = lnblk(s)
[..]
analyzevars: character array "character*8 phcod(57)" is considered as "character phcod(57,8)"; "intent(c)" is forced.
```

I would glad about any hint why this error occurs.

**Comment by mattip** [comment]
duplicate of gh-6308. The solution there is to replace `character*8 phcod` with `character(8) phcod`. Does this solve your error?

**Comment by HaoZeke** [comment]
Closing as duplicate, and one which is fixed by #19388.
